examples/cpp_execution/InferenceSetIOBufferExample.cpp

```diff
@@ -0,0 +1,511 @@
+        {
+            delete[] inputBuffer.buf;
+            inputBuffer.buf = new uint8_t[token_size_bytes];
+            inputBuffer.size = token_size_bytes;
+        }
+        std::copy_n(reinterpret_cast<const uint8_t *>(tokenVector.data()),
+                    token_size_bytes, inputBuffer.buf);
+
+        if (inputBuffer.size > token_size_bytes)
+        {
+            std::memset(inputBuffer.buf + token_size_bytes, 0,
+                        inputBuffer.size - token_size_bytes);
+        }
+    }
+
+    template <typename T>
+    [[nodiscard]] std::string qBufferToString(shQBufferWrapper wrappedBuf)
+    {
+        std::ostringstream strm;
+        auto rawBufPtr = wrappedBuf->getQBuffer().buf;
+        const T *bufferT = reinterpret_cast<const T *>(rawBufPtr);
+        int numT = wrappedBuf->getQBuffer().size / sizeof(T);
+        for (int i = 0; i < numT; i++)
+        {
+            strm << "[ " << i << " ] = " << bufferT[i] << "\n";
+        }
+        return strm.str();
+    }
+
+    /**
+     * Given buffer mapping instance, return true if this instance does not
+     * contain input or output buffers (e.g. it contains uninitialized or invalid)
+     * @param bufmap buffer mapping instance
+     * @return true if the buffer mapping instance does not container a valid buffer
+     */
+    [[nodiscard]] bool notInputOrOutput(const qaic::rt::BufferMapping &bufmap)
+    {
+        const std::initializer_list<QAicBufferIoTypeEnum> bufTypes{
+            BUFFER_IO_TYPE_INPUT, BUFFER_IO_TYPE_OUTPUT};
+        const auto func([type = bufmap.ioType](const auto v)
+                        { return v == type; });
+        return std::none_of(bufTypes.begin(), bufTypes.end(), func);
+    }
+
+    /**
+     * Given input and output buffers, release all heap allocated
+     * @param bufferMappings vector of BufferMapping
+     * @param inputBuffers vector of QBuffers - inputs
+     * @param outputBuffers vector of Qbuffers - outputs
+     * @param inputIdBuffers Qbuffers - input id
+     * @param positionIdBuffers Qbuffers - position id
+     */
+    void populateBuffersWithInputs(const std::vector<qaic::rt::BufferMapping> bufferMappings,
+                                   std::vector<QBuffer> &inputBuffers,
+                                   std::vector<QBuffer> &outputBuffers,
+                                   QBuffer &inputIdBuffer,
+                                   QBuffer &positionIdBuffer)
+    {
+        inputBuffers.clear();
+        outputBuffers.clear();
+        for (const auto &bufmap : bufferMappings)
+        {
+            if (notInputOrOutput(bufmap)) {
+                continue;
+            }
+            QBuffer buf{bufmap.size, new uint8_t[bufmap.size]};
+            populateVector(bufmap, buf, inputBuffers, outputBuffers);
+        }
+        // Filling last 2 index of inputBuffers with inputIds and positionIds
+        inputBuffers[inputBuffers.size() - 1] = positionIdBuffer;
```

Comment:
inputBuffers[- 1] should also work here.

Reply:
This will be invalid pointer in cpp as if the buffer size would be 0 it will be pointing to location which does not exist.